### PR TITLE
Ports Railings

### PR DESCRIPTION
**Pull request by Geevies**
Ported from Baystation.

As requested by the mappers.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/objects/structures/railing.dm [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+				to_chat(user, SPAN_WARNING("There's \a [occupied] in the way!"))
+				return
+
+			if(G.state >= GRAB_NECK)
+				if(user.a_intent == I_HURT)
+					visible_message(SPAN_DANGER("[G.assailant] slams [G.affecting]'s face against \the [src]!"))
+					playsound(get_turf(src), 'sound/effects/grillehit.ogg', 50, TRUE)
+					if(prob(30))
+						G.affecting.Weaken(5)
+					G.affecting.apply_damage(15, BRUTE, BP_HEAD)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/railing.dm]: This has no cooldown.

File: code/game/objects/structures/railing.dm [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+			return
+		user.visible_message(anchored ? "<span class='notice'>\The [user] begins unscrewing \the [src].</span>" : "<span class='notice'>\The [user] begins fastening \the [src].</span>" )
+		playsound(get_turf(src), W.usesound, 75, TRUE)
+		if(do_after(user, 10, src) && density)
+			to_chat(user, (anchored ? "<span class='notice'>You have unfastened \the [src] from the floor.</span>" : "<span class='notice'>You have fastened \the [src] to the floor.</span>"))
+			anchored = !anchored
+			update_icon()
+		return
+
+	if(W.force && (W.damtype == "fire" || W.damtype == "brute"))
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/railing.dm]:
```suggestion
	if(W.force && (W.damtype == BURN || W.damtype == BRUTE))
```

File: code/game/objects/structures/railing.dm (hunk 1 above)
Comment (Geevies) [reply code/game/objects/structures/railing.dm]: it do

File: code/game/objects/structures/railing.dm [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+		if(!density)
+			return TRUE
+		return FALSE
+	return TRUE
+
+/obj/structure/railing/attackby(var/obj/item/W, var/mob/user)
+	// Handle harm intent grabbing/tabling.
+	if(istype(W, /obj/item/grab) && user.Adjacent(src))
+		var/obj/item/grab/G = W
+		if(istype(G.affecting, /mob/living/carbon/human))
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/railing.dm]:
```suggestion
		if(ishuman(G.affecting))
```

File: code/game/objects/structures/railing.dm [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+		playsound(get_turf(src), W.usesound, 75, TRUE)
+		if(do_after(user, 10, src) && density)
+			to_chat(user, (anchored ? "<span class='notice'>You have unfastened \the [src] from the floor.</span>" : "<span class='notice'>You have fastened \the [src] to the floor.</span>"))
+			anchored = !anchored
+			update_icon()
+		return
+
+	if(W.force && (W.damtype == BURN || W.damtype == BRUTE))
+		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+		visible_message(SPAN_WARNING("\The [src] has been [LAZYLEN(W.attack_verb) ? pick(W.attack_verb) : "attacked"] with \the [W] by \the [user]!"))
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/railing.dm]: Add a sound here perhaps?

**Pull request merged**